### PR TITLE
Query distance with coordinate()

### PR DESCRIPTION
**Pull request by erneestoc**
Changed a Doctest to use updated data from Google's API.

I needed to query distance using coordinates. GoogleMaps.distance/3 can now use the coordinate() type to check distance between points.
```elixir
iex> {:ok, result2} = GoogleMaps.distance({27.5119772, -109.9409902}, 
                                          {19.4156207, -99.171256517})
iex> result2["rows"] \
      |> List.first() \
      |> Map.fetch!("elements") \ 
      |> List.first() \
      |> Map.fetch!("distance") \
      |> Map.fetch!("value")
1633309 # distance in meters between Mexico City and Ciudad Obregón
```

### CODE REVIEW

File: lib/google_maps.ex [hunk 1]
```diff
@@ -297,23 +297,39 @@ defmodule GoogleMaps do
...
+  def distance(origin, destination, options) when is_binary(destination) and is_binary(destination) do
     params = options
     |> Keyword.merge([origins: origin, destinations: destination])
 
     GoogleMaps.get("distancematrix", params)
   end
 
+  @spec distance(coordinate(), coordinate(), options()) :: Response.t()
+  def distance({lat1, lng1}, {lat2, lng2}, options) do
+    params = options
```
Comment (sntran) [inline lib/google_maps.ex]: For this one, can you just do `distance("#{lat1},#{lng1}", "#{lat2},#{lng2}")`?

File: lib/google_maps.ex [hunk 2]
```diff
@@ -297,23 +297,39 @@ defmodule GoogleMaps do
...
       iex> result["origin_addresses"]
       ["Place d'Armes, 78000 Versailles, France"]
       iex> [%{"elements" => [%{"distance" => distance}]}] = result["rows"]
       iex> distance["text"]
-      "23.3 km"
+      "23.8 km"
       iex> distance["value"]
-      23286
+      23763
+
```
Comment (sntran) [inline lib/google_maps.ex]: Please remove  one empty line here.

File: lib/google_maps.ex [hunk 3]
```diff
@@ -297,23 +297,39 @@ defmodule GoogleMaps do
...
-      "23.3 km"
+      "23.8 km"
       iex> distance["value"]
-      23286
+      23763
+      
+
+      # Distance from coordinate A to coordinate B
+      iex> {:ok, result2} = GoogleMaps.distance({27.5119772, -109.9409902}, {19.4156207, -99.171256517})
+      iex> result2["rows"] |> List.first() |> Map.fetch!("elements") |> List.first() |> Map.fetch!("distance") |> Map.fetch!("value")
```
Comment (sntran) [inline lib/google_maps.ex]: Please use multiline for readability, i.e.

```elixir
    iex> result2["rows"] 
    ...>   |> List.first()
    ...>   |> Map.fetch!("elements") 
    ...>   |> List.first() 
    ...>   |> Map.fetch!("distance")
    ...>   |> Map.fetch!("value")

File: lib/google_maps.ex [hunk 4]
```diff
@@ -297,23 +297,39 @@ defmodule GoogleMaps do
...
+      1633309
   """
+  def distance(origin, destination, options \\ [])
+
   @spec distance(address(), address(), options()) :: Response.t()
-  def distance(origin, destination, options \\ []) do
+  def distance(origin, destination, options) when is_binary(destination) and is_binary(destination) do
```
[the remark this answers is not in the document]
Comment (erneestoc) [reply lib/google_maps.ex]: 🙈

**Comment by erneestoc** [comment]
Thanks for your comments. Just uploaded requested changes.

**Comment by sntran** [comment]
👍  👍  👍

**Pull request merged**